src/Runner.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -622,22 +622,43 @@ private async Task<GitHubRunnerRegisterToken> GetJITRunnerTokenAsync(string gith
                 httpClient.DefaultRequestHeaders.UserAgent.AddRange(HostContext.UserAgents);
                 httpClient.DefaultRequestHeaders.Accept.ParseAdd("application/vnd.github.v3+json");
 
-                var response = await httpClient.PostAsync(githubApiUrl, new StringContent(string.Empty));
-
-                if (response.IsSuccessStatusCode)
+                int retryCount = 0;
```

Comment:
I would put the retry to include even create the HttpClient, since we might bind to a bad connection over load balancer, recreate new HttpClient can help us bouncing around...

ex:
https://github.com/actions/runner/blob/main/src/Runner.Worker/ActionManager.cs#L762

Reply:
Okay, makes sense, I included that too. Thanks for quick feedback.